Move admin route from "/" to "/admin"

myokoym opened the pull request.

Commit:
Move admin route from "/" to "/admin"

tricknotes commented on app/controllers/admin/sites_controller.rb:
```diff
@@ -33,7 +34,7 @@ def create
   def update
     if @site.update(site_params)
-      redirect_to @site, notice: 'Site was successfully updated.'
+      redirect_to [:admin, @site], notice: 'Site was successfully updated.'
```
Need to change `create` action too?

myokoym replied:
Thanks! I've fixed it.

myokoym replied:
あれ、違うところを直したみたいです。。。すみません、出直します。

myokoym replied:
あ、いいのか。修正したら↑のdiffが消えると思っていたのですが、
コメントの行を修正したわけではないのでこの場合は消えないんですね。

tricknotes commented on config/routes.rb:
```diff
@@ -1,7 +1,11 @@
 Rails.application.routes.draw do
-  root 'welcome#index'
```
It would be better some page for `root` exist, such as `sites#show`.

myokoym replied:
すみません、ここがちょっとわかりませんでした。。。
`root`をどこかに設定したほうがよさそう、という感じでしょうか？

tricknotes replied:
そうですね、いまだと Review Apps で表示したときにいきなり 404 ページが表示されてぎょっとなってしまうのでした…
これをなんとかできれば嬉しいなと思ってます！

myokoym replied:
わかりましたー！
ちなみに `sites#show` ってadminネームスペースのルーティングですよね。
`root`にネームスペースを指定するときってどうやるのがいいでしょうか…？

tricknotes replied:
あ、いまって公開用の `sites#show` ってまだ用意してないんでしたっけ。
そのうちトップページになるような画面が見えていると嬉しいなと思ってました。

単純にネームスペースを指定するのであれば、 `root 'admin/sites#show'` でどうでしょうか。

myokoym replied:
最初は `site_posts#index` にしようと思ったんですが、サブドメインがないと500エラーになるんですよね…

tricknotes replied:
nrhd-

myokoym replied:
> あ、いまって公開用の `sites#show` ってまだ用意してないんでしたっけ。
> そのうちトップページになるような画面が見えていると嬉しいなと思ってました。

あ、なるほど、公開用の`sites#show`ですか。まだ用意してなかったです…

> 単純にネームスペースを指定するのであれば、 `root 'admin/sites#show'` でどうでしょうか。

はっ、スラッシュ区切り、普通ですね…ありがとうございます :sob: 
ひとまず `admin/sites#index` にしておこうと思います。

Commit:
Use /admin route for sites#create

Commit:
Set root route to prevent 404 error

Commit:
Improve root route

myokoym commented:
全面的に同意です！修正しました。

tricknotes commented:
:dizzy:

mtsmfm commented on app/controllers/admin/posts_controller.rb:
```diff
@@ -1,5 +1,6 @@
-class PostsController < ApplicationController
+class Admin::PostsController < ApplicationController
   before_action :set_post, only: [:show, :edit, :update, :destroy]
+  layout 'admin'
```
Why don't you introduce `Admin::ApplicationController` to DRY `layout` declaration?

``` ruby
class Admin::ApplicationController < ActionController::Base
  layout 'admin'
end

class Admin::PostsController < Admin::ApplicationController
end
```

myokoym replied:
Thanks, I think it's better way. Thanks @okkez.

okkez committed:
Add Admin::ApplicationController

okkez committed:
Use Admin::ApplicationController instead of ApplicationController

In previous version, we set layout in each class under Admin namespace.
In this version, we set layout in Admin::ApplicationController and
controllers that are under Admin namespace inherit it.

mtsmfm commented:
:shipit:
